plugins/modules/foreman_os_default_template.py

```diff
@@ -107,8 +107,8 @@ def main():
     entity_dict['template_kind'] = module.find_resource_by_name('template_kinds', entity_dict['template_kind'], thin=True)
 
     scope = {'operatingsystem_id': entity_dict['operatingsystem']['id']}
-    search = 'template_kind_id={}'.format(entity_dict['template_kind']['id'])
-    entity = module.find_resource('os_default_templates', search, params=scope, failsafe=True)
+    entities = module.list_resource('os_default_templates', params=scope)
```

Comment:
can you add a comment, linking to https://projects.theforeman.org/issues/27722 for the explanation